src/ledger/ledger-radix-babylon.ts

```diff
@@ -0,0 +1,87 @@
+import Transport from '@ledgerhq/hw-transport'
+import { errAsync, ok, ResultAsync } from 'neverthrow'
+import { LedgerPublicKeyRequest } from './schemas'
+
+export const encodeBip32 = (hdPath: string) => {
+  const path = hdPath.slice(2).split('/')
+  const length = `00${(path.length & 0xff).toString(16)}`.slice(-2)
+
+  const parts = path
+    .map((value) => 0x80000000 + parseInt(value, 10))
+    .map((value) => value.toString(16))
+    .join('')
+
+  return `${length}${parts}`
+}
+
+export enum LedgerErrorResponse {
+  UnlockDevice = '5515',
+  OpenRadixApp = 28161,
+}
+
+export enum LedgerInstructionCode {
+  GetDeviceModel = 0x11,
+  GetDeviceId = 0x12,
+  GetPubKeyEd25519 = 0x21,
+  GetPubKeySecp256k1 = 0x31,
+}
+
+export const errorResponses: Record<string, string> = {
+  [LedgerErrorResponse.UnlockDevice]:
+    'Please unlock Ledger Device and try again',
+  [LedgerErrorResponse.OpenRadixApp]:
+    'Please open Radix Babylon app in your Ledger device and try again',
+}
+
+export class LedgerRadixBabylon {
+  public constructor(private transport: Transport) {}
+
+  public getDeviceInfo() {
+    return this.exchange(LedgerInstructionCode.GetDeviceId).andThen(
+      (deviceId) =>
+        this.exchange(LedgerInstructionCode.GetDeviceModel).map((model) => ({
+          deviceId,
+          model: Number(model),
+        }))
+    )
+  }
+
+  public getPublicKey(
+    params: Omit<LedgerPublicKeyRequest['requestPublicKey'], 'deviceName'>
+  ) {
+    return this.exchange(LedgerInstructionCode.GetDeviceId).andThen(
+      (deviceId) => {
+        if (deviceId !== params.deviceId) {
+          return errAsync(
+            "Device doesn't match. Make sure you connected correct Ledger device"
+          )
+        }
+        const bip32Data = encodeBip32(params.hdPathNumber)
+        const dataLength = Math.floor(bip32Data.length / 2).toString(16)
+        return this.exchange(
+          params.curve === 'ed25519'
+            ? LedgerInstructionCode.GetPubKeyEd25519
+            : LedgerInstructionCode.GetPubKeySecp256k1,
+          `${dataLength}${bip32Data}`
+        )
+      }
+    )
+  }
+
+  private exchange(command: LedgerInstructionCode, data = '') {
+    return ResultAsync.fromPromise<Buffer, string>(
+      this.transport.exchange(
+        Buffer.from(`aa${command.toString(16)}0000${data}`, 'hex')
+      ),
+      (err: any) => {
+        if (err?.name === 'LockedDeviceError') {
+          return errorResponses[LedgerErrorResponse.UnlockDevice]
+        }
+
+        return errorResponses[err?.statusCode] || err.message
+      }
+    ).andThen((buffer) =>
+      ok(buffer.slice(0, buffer.length - 2).toString('hex'))
```

Comment:
`The signature '(start?: number | undefined, end?: number | undefined): Buffer' of 'buffer.slice' is deprecated.`